### PR TITLE
Update showHiddenChannels to work with Forums

### PR DESCRIPTION
**Pull request by nmsturcke**
Update the showHiddenChannels plugin to work with Forums.

This PR is based on #234 as it uses the `Badge` component.

## Before 
image

## After
image

### CODE REVIEW

**Comment by Vendicated** [comment]
Please rebase!

**Comment by nmsturcke** [comment]
idk if that worked tbh

File: src/plugins/showHiddenChannels.tsx [hunk 1]
```diff
@@ -117,16 +118,17 @@ export default definePlugin({
         const isHidden = this.isHiddenChannel(channel);
         // check for type again, otherwise it would show it for hidden stage channels
         if (channel.type === 0 && isHidden) {
-            const lastMessageDate = channel.lastMessageId ? new Date(SnowflakeUtils.extractTimestamp(channel.lastMessageId)).toLocaleString() : null;
+            const lastMessageDate = channel.lastActiveTimestamp ? new Date(channel.lastActiveTimestamp).toLocaleString() : null;
```
Comment (Vendicated) [inline src/plugins/showHiddenChannels.tsx:121]: undo this change please

File: src/plugins/showHiddenChannels.tsx [hunk 2]
```diff
@@ -97,7 +98,7 @@ export default definePlugin({
     shouldShow(channel, category, isMuted) {
         if (!this.isHiddenChannel(channel)) return false;
         if (!category) return false;
-        if (channel.type === 0 && category.guild?.hideMutedChannels && isMuted) return false;
+        if ((channel.type === 0 || channel.type === 15) && category.guild?.hideMutedChannels && isMuted) return false;
```
Comment (Vendicated) [inline src/plugins/showHiddenChannels.tsx:101]: please make this a constant instead so it's easy to tell what 15 means

Comment (ArjixWasTaken) [reply src/plugins/showHiddenChannels.tsx:101]: Or even better, use an enum

Does the discord-tyoes module not have it?

File: src/plugins/showHiddenChannels.tsx [hunk 3]
```diff
@@ -25,13 +25,28 @@ import { ModalContent, ModalFooter, ModalHeader, ModalRoot, ModalSize, openModal
...
+    GROUP_DM = 3,
+    GUILD_CATEGORY = 4,
+    GUILD_ANNOUNCEMENT = 5,
+    ANNOUNCEMENT_THREAD = 10,
+    PUBLIC_THREAD = 11,
+    PRIVATE_THREAD = 12,
+    GUILD_STAGE_VOICE = 13,
+    GUILD_DIRECTORY = 14,
+    GUILD_FORUM = 15
+}
```
Comment (Vendicated) [inline src/plugins/showHiddenChannels.tsx:41]: could you remove the unused ones please? Also some parts of the code still use hardcoded number, please update those

File: src/components/Badge.tsx [hunk 4]
```diff
@@ -21,6 +21,7 @@ export function Badge({ text, color }): JSX.Element {
         <div className="vc-plugins-badge" style={{
             backgroundColor: color,
             justifySelf: "flex-end",
+            alignSelf: "center",
```
Comment (Vendicated) [inline src/components/Badge.tsx:24]: what's this for?

**Comment by Vendicated** [comment]
thank you! Superseded by #446
